### PR TITLE
Crash reporter/dev

### PR DESCRIPTION
**Pull request by jameskeith**
Crash Reporter 1.0.0
 - Saves crash data on uncaught exceptions, sends crash data on next activity/app launch
 - Test coverage including jacoco
 - Basic app crash example
 - ProGuard implementation (disabled) ready for when Android Gradle plugin is up to 4.1.1 to support Kotlin extensions.

### CODE REVIEW

File: crashreporter/build.gradle [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+android {
+    compileSdkVersion 29
+    buildToolsVersion "29.0.3"
+
+    defaultConfig {
+        minSdkVersion 21
+        targetSdkVersion 29
+        versionCode 1
+        versionName "1.0"
+        buildConfigField 'String', 'TAG', "\"Tealium-CollectDispatcher-$version\""
```
Comment (jonathanwong) [inline crashreporter/build.gradle:18]: Typo: `CollectDispatcher` -> `CrashReporter`.

**Review by jonathanswong — APPROVED** [review]

File: crashreporter/build.gradle [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+android {
+    compileSdkVersion 29
+    buildToolsVersion "29.0.3"
+
+    defaultConfig {
+        minSdkVersion 21
+        targetSdkVersion 29
+        versionCode 1
+        versionName "1.0"
```
Comment (tamayok) [inline crashreporter/build.gradle:17]: Should this be "$version"?

File: crashreporter/src/main/java/com/tealium/crashreporter/TealiumConfigCrashReporter.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import com.tealium.core.ModuleFactory
+import com.tealium.core.Modules
+import com.tealium.core.Tealium
+import com.tealium.core.TealiumConfig
+
+/**
+ * Returns the Crash Reporter factory.
+ */
+val Modules.CrashReporter: ModuleFactory
+    get() = com.tealium.crashreporter.CrashReporter
```
Comment (tamayok) [inline crashreporter/src/main/java/com/tealium/crashreporter/TealiumConfigCrashReporter.kt:12]: Do we typically add this bit in the Config extension? Isn't it usually added to the entry point of the module factory - `CrashReporter.kt`?

File: crashreporter/src/main/java/com/tealium/crashreporter/internal/Crash.kt [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+            } catch (ex: JSONException) {
+                Logger.prod(BuildConfig.TAG, "Error getting Thread data: ${ex.message ?: ""}")
+            }
+
+            array.put(threadData.toString())
+
+            return array.toString()
+        }
+
+        internal fun getStackData(crash: Crash, truncateStackTrace: Boolean): JSONArray {
```
Comment (tamayok) [inline crashreporter/src/main/java/com/tealium/crashreporter/internal/Crash.kt:51]: Could be private? Only called from within the data class.

**Review by tamayok — APPROVED** [review]
Looks good! Good work.

**Pull request merged**